File: Top Interview 150/Two Pointers/125_Valid Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string ans = "";
        for (auto ch : s) {
            if (isalnum(ch)){
                ans += isalpha(ch) ? tolower(ch) : ch;
            }
        }
        for(int i=0; i < ans.length()/2; i++) {
            if (ans[i] != ans[ans.length() -1 - i]) {
                return false;
            }
        }
        return true;
    }
};

File: Top Interview 150/Two Pointers/125_Valid Palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        # convert to lowercase
        s = s.lower()
        # removes spaces and special characters
        s = ''.join(letter for letter in s if letter.isalnum())

        a = 0
        
        for i in range(int(len(s)/2)):
            if s[i] == s[-1-i]:
                a += 1

        if a == int(len(s)/2):
            return True
        else:
            return False